File: trip-backend/src/models/agent_step.py
"""AgentStep model (agent_steps table)"""

from sqlalchemy import Column, Integer, String, JSON, ForeignKey, Text, Index
from sqlalchemy.orm import relationship

from src.models.base import Base, BaseModelWithoutUpdatedAt


class AgentStep(Base, BaseModelWithoutUpdatedAt):
    """AgentStep model matching Prisma schema (no updated_at field)"""
    
    __tablename__ = "agent_steps"
    __table_args__ = (
        Index("idx_agent_steps_msg_step", "message_id", "step"),
        {"comment": "Agent步骤表"},
    )
    
    # Fields (id and created_at inherited from BaseModelWithoutUpdatedAt)
    message_id = Column(
        "message_id",
        Integer,
        ForeignKey("messages.id", ondelete="CASCADE"),
        nullable=False,
        comment="消息ID（外键）"
    )
    step = Column(
        Integer,
        nullable=False,
        comment="步骤序号"
    )
    type = Column(
        String(20),
        nullable=False,
        comment="步骤类型"
    )
    name = Column(
        String(100),
        nullable=True,
        comment="步骤名称"
    )
    args = Column(
        JSON,
        nullable=True,
        comment="参数（JSON）"
    )
    output = Column(
        Text,
        nullable=True,
        comment="输出"
    )
    duration_ms = Column(
        "duration_ms",
        Integer,
        nullable=True,
        comment="耗时（毫秒）"
    )
    error = Column(
        Text,
        nullable=True,
        comment="错误信息"
    )
    
    # Relationships
    message = relationship("Message", back_populates="steps")
    
    def __repr__(self):
        return f"<AgentStep(id={self.id}, step={self.step}, type={self.type})>"

File: trip-backend-java/src/main/java/com/trip/backend/domain/entity/AgentStep.java
package com.trip.backend.domain.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.OffsetDateTime;
import java.util.Map;

/**
 * Agent 步骤实体（对应 Python models/agent_step.py）
 * - args 为 JSONB 列
 * - 无 updated_at
 * - ondelete CASCADE（message_id FK）
 */
@Entity
@Table(name = "agent_steps", indexes = {
    @Index(name = "idx_agent_steps_message_id", columnList = "message_id, step")
})
@Getter
@Setter
public class AgentStep {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long messageId;

    @Column(nullable = false)
    private Integer step;

    @Column(nullable = false, length = 100)
    private String type;

    @Column(nullable = false, length = 200)
    private String name;

    // JSONB 列
    @JdbcTypeCode(SqlTypes.JSON)
    @Column(columnDefinition = "jsonb")
    private Map<String, Object> args;

    @Lob
    private String output;

    @Column
    private Long durationMs;

    @Column(length = 500)
    private String error;

    @Column(nullable = false)
    private OffsetDateTime createdAt;

    // 无 updated_at

    protected AgentStep() {}

    @PrePersist
    protected void onCreate() {
        this.createdAt = OffsetDateTime.now();
    }
}
